### PR TITLE
Add cmakefiles for remaining projects

### LINKED ISSUE #1052: cmake build not building all binaries
List of all binaries produced by msbuild build that are missing from cmake build.
- [ ] bindmonitor.sys
- [ ] bindmonitor_ringbuf.sys
- [ ] bindmonitor_ringbuf_um.dll
- [ ] bindmonitor_tailcall.sys
- [ ] bindmonitor_tailcall_um.dll
- [ ] bindmonitor_um.dll
- [ ] bpf.sys
- [ ] bpf_call.sys
- [ ] bpf_call_um.dll
- [ ] bpf_um.dll
- [ ] bpftool_tests.exe
- [ ] cgroup_sock_addr.sys
- [ ] cgroup_sock_addr_um.dll
- [ ] cilium_tests.exe
- [ ] concrt140d.dll
- [ ] decap_permit_packet.sys
- [ ] decap_permit_packet_um.dll
- [ ] divide_by_zero.sys
- [ ] divide_by_zero_um.dll
- [ ] droppacket.sys
- [ ] droppacket_um.dll
- [ ] droppacket_unsafe.sys
- [ ] droppacket_unsafe_um.dll
- [ ] encap_reflect_packet.sys
- [ ] encap_reflect_packet_um.dll
- [ ] fuzz.exe
- [ ] invalid_helpers.sys
- [ ] invalid_helpers_um.dll
- [ ] invalid_maps1.sys
- [ ] invalid_maps1_um.dll
- [ ] invalid_maps2.sys
- [ ] invalid_maps2_um.dll
- [ ] map.sys
- [ ] map_in_map.sys
- [ ] map_in_map_um.dll
- [ ] map_in_map_v2.sys
- [ ] map_in_map_v2_um.dll
- [ ] map_reuse.sys
- [ ] map_reuse_2.sys
- [ ] map_reuse_2_um.dll
- [ ] map_reuse_um.dll
- [ ] map_um.dll
- [ ] msvcp140_1d.dll
- [ ] msvcp140_2d.dll
- [ ] msvcp140d.dll
- [ ] msvcp140d_atomic_wait.dll
- [ ] msvcp140d_codecvt_ids.dll
- [ ] printk.sys
- [ ] printk_legacy.sys
- [ ] printk_legacy_um.dll
- [ ] printk_um.dll
- [ ] printk_unsafe.sys
- [ ] printk_unsafe_um.dll
- [ ] reflect_packet.sys
- [ ] reflect_packet_um.dll
- [ ] sample_ebpf_ext.sys
- [ ] sample_ext_app.exe
- [ ] socket_tests.exe
- [ ] sockops.sys
- [ ] sockops_um.dll
- [ ] tail_call.sys
- [ ] tail_call_bad.sys
- [ ] tail_call_bad_um.dll
- [ ] tail_call_map.sys
- [ ] tail_call_map_um.dll
- [ ] tail_call_multiple.sys
- [ ] tail_call_multiple_um.dll
- [ ] tail_call_um.dll
- [ ] test_sample_ebpf.sys
- [ ] test_sample_ebpf_um.dll
- [ ] test_utility_helpers.sys
- [ ] test_utility_helpers_um.dll
- [ ] ucrtbased.dll
- [ ] vccorlib140d.dll
- [ ] vcruntime140_1d.dll
- [ ] vcruntime140d.dll

### PR DESCRIPTION
**Pull request by shankarseal**
## Description

Add cmakefiles for remaining projects. Fixes #1052 .

## Testing

Local build.

## Documentation

None.

### CODE REVIEW

File: scripts/Convert-BpfToNative.ps1 [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 param([parameter(Mandatory = $true)] [string] $ProgramName,
     [parameter(Mandatory = $false)] [string] $SolutionDir = $Pwd,
+    [parameter(Mandatory = $false)] [string] $OutputDir = $Pwd,
```
Comment (Alan-Jowett) [inline scripts/Convert-BpfToNative.ps1:6]: Need to chat about this. This will conflict with #1097

**Review by Alan-Jowett — APPROVED** [review]

File: .github/workflows/cicd.yml [hunk 2]
```diff
@@ -38,15 +38,15 @@ jobs:
...
+      build_artifact: Build-x64-vcxproj
       generate_release_package: true
 
   cmake:
     # Always run this job.
     if: github.event_name == 'schedule' || github.event_name == 'pull_request' || github.event_name == 'push'
     uses: ./.github/workflows/reusable-cmake-build.yml
     with:
-      build_artifact: Build-x64-cmake
+      build_artifact: Build-x64
```
Comment (dthaler) [inline .github/workflows/cicd.yml:49]: Why the change on this line?  Where does the string come from?

**Review by dthaler — APPROVED** [review]

File: scripts/Convert-BpfToNative.ps1 (hunk 1 above)
Comment (shankarseal) [reply scripts/Convert-BpfToNative.ps1:6]: Lets get this one in first, then we will fight with 1097.

File: .github/workflows/cicd.yml (hunk 2 above)
Comment (shankarseal) [reply .github/workflows/cicd.yml:49]: This is to test if the tests work with the output of the cmake build and not the regular build. In a subsequent PR all the checked in vcxproj files and this CI task itself will be removed.

**Review by Alan-Jowett — APPROVED** [review]
